packages/admin/admin/src/form/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,164 @@
+import { Delete, Select } from "@comet/admin-icons";
+import { Button, Chip, ComponentsOverrides, FormHelperText, IconButton, Theme } from "@mui/material";
+import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import clsx from "clsx";
+import * as React from "react";
+import { Accept, useDropzone } from "react-dropzone";
+import { FieldRenderProps } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+import { PrettyBytes } from "../helpers/PrettyBytes";
+
+export type FinalFormFileUploadClassKey = "root" | "droppableArea" | "droppableAreaCaption" | "fileList" | "fileListItem" | "disabled" | "error";
+
+const styles = ({ palette }: Theme) => {
+    return createStyles<FinalFormFileUploadClassKey, FinalFormFileUploadProps>({
+        root: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "10px",
+            alignSelf: "stretch",
+            minWidth: "350px",
+        },
+        droppableArea: {
+            display: "flex",
+            height: "80px",
+            padding: "10px",
+            justifyContent: "center",
+            alignItems: "center",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            border: `1px dashed ${palette.text.secondary}`,
+            cursor: "pointer",
+            minWidth: "250px",
+            "&$disabled": {
+                cursor: "default",
+            },
+            "&$error": {
+                border: `1px dashed ${palette.error.main}`,
+            },
+        },
+        droppableAreaCaption: {
+            color: palette.text.secondary,
+            fontSize: "14px",
+            fontStyle: "normal",
+            fontWeight: "300",
+            lineHeight: "20px",
+        },
+        fileList: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "2px",
+            alignSelf: "stretch",
+        },
+        fileListItem: {
+            display: "flex",
+            height: "35px",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+        },
+        disabled: {},
+        error: {},
+    });
+};
+
+export interface FinalFormFileUploadProps extends FieldRenderProps<File[], HTMLInputElement> {
```

Comment:
For single select, the field value shouldn't be an array, but a single file: `FieldRenderProps<File | File[]>, HTMLInputElement>`. Please ensure that this works correctly.

Reply:
I have a problem with the array functions here and I'm not sure how to solve it
![Screenshot 2024-01-08 at 15 18 09](https://github.com/vivid-planet/comet/assets/72875628/3e72a416-a9fd-4549-9548-ba0648e837b2)

Reply:
You'll need to have some `Array.isArray` checks to tackle this. We can pair program on it together if you want

Reply:
OK I added it now and it works, but I'm not sure - could you check please? https://github.com/vivid-planet/comet/pull/1461/commits/02dc9854eee71aac3a0ae89aa66f7b520553d215

Reply:
Looks good! Added two suggestions to reduce complexity in my last review 🙂

Reply:
makes it much easier, thank you!